File: bits/bit_fields.c
#include <stdio.h>

#define A 0x01
#define B 0x02
#define C 0x04

void say_hello(int);

int main(void)
{
    say_hello(A | C);
    say_hello(A | B);

    return 0;
}

void say_hello(int opts)
{
    if (opts & A)
        printf("A ");

    if (opts & B)
        printf("B ");

    if (opts & C)
        printf("C ");

    printf("\n");
}

File: learn/src/bits/BitFields.java
package bits;

public final class BitFields {
    private BitFields() {}

    public static final int OPT_A = 0x01;
    public static final int OPT_B = 0x02;
    public static final int OPT_C = 0x04;

    public static void optionsFrom(int opt) {
        if ((opt & OPT_A) > 0) {
            System.out.println("OPT_A");
        }

        if ((opt & OPT_B) > 0) {
            System.out.println("OPT_B");
        }

        if ((opt & OPT_C) > 0) {
            System.out.println("OPT_C");
        }
    }

    public static void main(String[] args) {
        BitFields.optionsFrom(OPT_B | OPT_C);
    }
}
